custom/succeed/reports/patient_task_list.py

```diff
@@ -1,66 +1,50 @@
-from custom.succeed.utils import is_succeed_admin, has_any_role, SUCCEED_CM_APPNAME, get_app_build
-from casexml.apps.case.models import CommCareCase
+from custom.succeed.reports import EMPTY_FIELD, INPUT_DATE_FORMAT, OUTPUT_DATE_FORMAT, \
+    CM_APP_UPDATE_VIEW_TASK_MODULE, CM_UPDATE_TASK, TASK_RISK_FACTOR, TASK_ACTIVITY
+from custom.succeed.utils import SUCCEED_CM_APPNAME, get_app_build
 from dimagi.utils.decorators.memoized import memoized
 
 
-class PatientTaskListReportDisplay(CaseDisplay):
-    def __init__(self, report, case_dict):
-
-        next_visit = VISIT_SCHEDULE[0]
-        last_inter = None
-        for action in case_dict['actions']:
-            if action['xform_xmlns'] in LAST_INTERACTION_LIST:
-                last_inter = action
+class PatientTaskListReport(SqlTabularReport, CustomProjectReport, ProjectReportParametersMixin):
+    name = ugettext_noop('Patient Tasks')
+    slug = 'patient_task_list'
+    table_name = 'fluff_UCLATaskActivityFluff'
+    fields = ['custom.succeed.fields.ResponsibleParty',
+              'custom.succeed.fields.PatientName',
+              'custom.succeed.fields.TaskStatus']
 
-        for visit_key, visit in enumerate(VISIT_SCHEDULE):
-            for key, action in enumerate(case_dict['actions']):
-                if visit['xmlns'] == action['xform_xmlns']:
-                    try:
-                        next_visit = VISIT_SCHEDULE[visit_key + 1]
-                        del case_dict['actions'][key]
-                        break
-                    except IndexError:
-                        next_visit = 'last'
-        self.next_visit = next_visit
-        if last_inter:
-            self.last_interaction = last_inter['date']
-        self.domain = report.domain
-        self.app_dict = get_cloudcare_app(self.domain, SUCCEED_CM_APPNAME)
+    def __init__(self, request, base_context=None, domain=None, **kwargs):
+        super(PatientTaskListReport, self).__init__(request, base_context=base_context, domain=domain, **kwargs)
+        self.app_dict = get_cloudcare_app(domain, SUCCEED_CM_APPNAME)
         self.latest_build = get_app_build(self.app_dict)
-        super(PatientTaskListReportDisplay, self).__init__(report, case_dict)
-
-    def get_property(self, key):
-        if key in self.case:
-            return self.case[key]
-        else:
-            return EMPTY_FIELD
 
-    def get_link(self, url, field):
+    def get_link(self, url, field, doc_id):
         if url:
-            return html.mark_safe("<a class='ajax_dialog' href='%s' target='_blank'>%s</a>" % (url, html.escape(field)))
+            try:
+                return html.mark_safe("<a class='ajax_dialog' href='{0}' target='_blank'>{1}</a>".format(
+                    url, html.escape(field)))
+            # added because we have an one case where in name we have a char \u2022 - middle dot
+            except UnicodeEncodeError:
```

Comment:
The try/except should not be needed. It looks like the problem here is your template string is a byte string, not Unicode. Have you tried this:
```
return html.mark_safe(u"<a class...
# note ---------------^
```